portfolio/src/main/webapp/charts.js

```diff
@@ -56,24 +67,25 @@ function fillCharts(){
 
         var Q1Array = [];
         var Q2Array = [];
-
-        for(i=0; i<properties.length; i++){
-            var row = i;
-            var question = properties[row][2];
-            if(question == "Q1"){
-                Q1Array.push(properties[row]);
-            }
-            else{
-                Q2Array.push(properties[row]);
+        if(properties.length != 0){
+
+            for(i=0; i<properties.length; i++){
```

Comment:
This is redundant.  If properties.length is 0 the for loop won't run

Reply:
But, I also don't want to sort and prepare the array to go into charts, which method calls are outside of the for-loop, if the array has nothing in it.